client/rest/src/plugins/metadata/metadataRoutes.js

```diff
@@ -62,5 +66,48 @@ module.exports = {
 				next();
 			});
 		});
+
+		server.get('/metadata/metal/:metalId', async (req, res, next) => {
+			const sendData = (data, mimeType, fileName, text, download) => routeUtils.createSender('content').sendData(res, next)(
+				data,
+				mimeType,
+				fileName,
+				text,
+				download
+			);
+			const setParams = (text, mimeType, fileName) => {
+				let seal;
+				try {
+					seal = MetalSeal.parse(text);
+				} catch {
+					// ignore
+				}
```

Comment:
i meant to ask, is this simply defensive programming. or do you have scenarios in mind where seal can be in wrong format?

if invalid data is expected, i'd change parse (throw exception) to tryParse, which could return undefined in case of error. i am also unsure in the case where the data is unparesable if that should be ignored in the route or result in a 400 error?

Reply:
sorry, i understood.
tryParse is better. maybe returns is object (isParsed: boolean, value: MetalSeal | string) is better.

```js
/**
* Try parses a JSON string into a MetalSeal object.
* The JSON string should represent an array with the schema, length, mimeType (if exists), name (if exists), and comment (if exists).
* @param {string} json - The JSON string to parse.
* @returns {object} - if could parse, return isParsed is true and a new MetalSeal object created from the parsed JSON string.
* Otherwise, isParsed is false and the return value is original.
* - isParsed: boolean
* - value: MetalSeal | string
*/
static tryParse(json) {
let parsedObject;
try {
	parsedObject = JSON.parse(json);
} catch {
	return { isParsed: false, value: json };
}
if (!Array.isArray(parsedObject)
	|| !this.isHead(parsedObject)
	|| !MetalSeal.COMPAT.includes(parsedObject[0])
)
	return { isParsed: false, value: json };

return {
	isParsed: true,
	value: new MetalSeal(
		parsedObject[1],
		parsedObject[2] ?? undefined,
		parsedObject[3] ?? undefined,
		parsedObject[4] ?? undefined,
		parsedObject[0]
	)
};
}
```